server/tests/user.spec.js

```diff
@@ -0,0 +1,278 @@
+          done();
+        });
+    });
+
+    it('Should fail for incorrect username and/or password', (done) => {
+      request.post('/login')
+        .send({ username: 'incorrect user',
+          password: 'incorrect password' })
+        .expect(401)
+        .end((err, res) => {
+          expect(res.body.message).to.equal('Login Failed');
+          done();
+        });
+    });
+
+    it('Should log a user out', (done) => {
+      request.post('/logout')
+        .expect(200)
+        .end((err, res) => {
+          expect(res.body.message).to.equal('Successful logout');
+          done();
+        });
+    });
+  });
+
+  describe('Post: (/users) - Create a user', () => {
+    it('Should create a unique user', (done) => {
+      request.post('/users')
+        .send(adminUserParam)
+        .expect(400)
+        .end((err, res) => {
+          expect(res.body.errors[0].type).to.equal('unique violation');
+          done();
+        });
+    });
+
+    it('Should test if new user has first name and last name', (done) => {
+      request.post('/users')
+        .set({ 'x-access-token': adminToken })
+        .send(regularUserParam)
+        .expect(200)
+        .end((err, res) => {
+          regularUser = res.body.user;
+          regularToken = res.body.token;
+          expect(regularUser.firstname).to.exist;
+          expect(regularUser.lastname).to.exist;
+          done();
+        });
+    });
+
+    it('Should ensure that new user has a role', (done) => {
+      request.post('/users')
+        .set({ 'x-access-token': adminToken })
+        .expect(201)
+        .end((err, res) => {
+          expect(regularUser).to.have.property('RoleId');
+          expect(regularUser.RoleId).to.equal(2);
+          done();
+        });
+    });
+
+    it('Should ensure user cannot be created if one of email or password is lacking.', (done) => {
+      testUserParam.email = null;
+      testUserParam.password = null;
+      request.post('/users')
+        .set({ 'x-access-token': adminToken })
+        .send(testUserParam)
+        .expect(500)
+        .end((err, res) => {
+          expect(res.body.errors[0].message).to.equal('email cannot be null');
+          expect(res.body.errors[1].message).to.equal('password cannot be null');
+          done();
+        });
+    });
+  });
+
+  describe('Get: (/users/) - Get a user', () => {
+    it('should not return a user if id is invalid', (done) => {
+      request.get('/users/123')
+        .set({ 'x-access-token': adminToken })
+        .expect(404)
+        .end((err, res) => {
+          expect(res.body.message).to.be.equal('User Not Found');
+          done();
+        });
+    });
+
+    it('Should return all users', (done) => {
+      request.get('/users')
+        .set({ 'x-access-token': adminToken })
+        .expect(200)
+        .end((err, res) => {
+          expect(Array.isArray(res.body)).to.equal(true);
+          expect(res.body[0]).to.have.property('id');
```

Comment:
What if you have an object that contains the fields you want and you loop through the object while running the expect on each property. That should reduce the lines of code here